benchmark_v2/benchmark_framework.py

```diff
@@ -0,0 +1,1181 @@
+            self.logger.debug(f"SDPA backend '{self.backend_name}' requested but not using kernel context (backend={self.backend})")
+        return self
+        
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.context is not None:
+            try:
+                self.context.__exit__(exc_type, exc_val, exc_tb)
+            except Exception as e:
+                if self.logger:
+                    self.logger.warning(f"Error exiting SDPA context: {e}")
+        return False
+
+
+class AbstractModelBenchmark(ABC):
+    """Abstract base class for model benchmarks."""
+    
+    def __init__(self, logger: logging.Logger):
+        self.logger = logger
+        self.model = None
+        self.tokenizer = None
+        self.device = None
+        self.scenarios = {}  # Map of scenario_name -> BenchmarkScenario
+        
+    @abstractmethod
+    def create_scenarios(self, **kwargs) -> Dict[str, 'BenchmarkScenario']:
+        """Create and return a dictionary of benchmark scenarios."""
+        pass
+        
+    @abstractmethod
+    def setup_model(self, config: BenchmarkConfig) -> None:
+        """Setup the model for benchmarking with the given configuration."""
+        pass
+    
+    @abstractmethod
+    def cleanup_model(self) -> None:
+        """Cleanup model resources."""
+        pass
+    
+    @abstractmethod
+    def measure_time_to_first_token(self, config: BenchmarkConfig) -> float:
+        """Measure time to first token generation."""
+        pass
+    
+    @abstractmethod
+    def measure_latency(self, config: BenchmarkConfig) -> TimingResult:
+        """Measure full generation latency and compute tokens/sec."""
+        pass
+    
+    def prepare_inputs(self, config: BenchmarkConfig) -> Any:
+        """Prepare inputs for the model. Override if needed."""
+        return None
+    
+    def get_scenarios(self, **kwargs) -> Dict[str, 'BenchmarkScenario']:
+        """Get benchmark scenarios. Creates them if they don't exist."""
+        if not self.scenarios:
+            self.scenarios = self.create_scenarios(**kwargs)
+        return self.scenarios
+
+
+class ModelBenchmark(AbstractModelBenchmark):
+    """
+    Base class for HuggingFace Transformers model benchmarks.
+    
+    This class provides common scenario creation logic and handles the standard
+    patterns for eager, compiled, and kernelized execution variants with different
+    attention implementations and SDPA backends.
+    """
+    
+    def __init__(self, logger: logging.Logger):
+        super().__init__(logger)
+        self.inputs = None
+        self.compiled_model = None
+        self.past_key_values = None
+        self.config = None
+        self._default_prompt = "Why dogs are so cute?"
+        
+    @property
+    def default_prompt(self) -> str:
+        """Default prompt for text generation. Override in subclasses if needed."""
+        return self._default_prompt
+    
+    @property
+    def model_type(self) -> str:
```

Comment:
What's the purpose of `model_type`?

Reply:
Leftover from another iteration, removed it. :)